src/providers/events.js

```diff
@@ -96,7 +106,46 @@ export function createEventsObject(openmct, parentKey, namespace) {
         }
     };
 
-    return eventsObject;
+    return baseEventObject;
+}
+
+export function createEventSeverityObjects(openmct, parentEventObject, namespace) {
+    const childSeverityObjects = [];
+    for (const severity of SEVERITY_LEVELS) {
+        const severityIdentifier = {
+            key: `${parentEventObject.identifier.key}.${severity}`,
+            namespace
+        };
+
+        const severityName = `${parentEventObject.name}: ${severity}`;
+
+        const severityEventObject = createEventObject(
+            openmct,
+            parentEventObject.identifier.key,
+            namespace,
+            severityIdentifier,
+            severityName,
+            OBJECT_TYPES.EVENT_SPECIFIC_SEVERITY_OBJECT_TYPE
+        );
+
+        childSeverityObjects.push(severityEventObject);
+    }
+
+    return childSeverityObjects;
+}
+
+export async function getEventSources(url, instance) {
+    const eventSourceURL = `${url}/api/archive/${instance}/events/sources`;
+    const eventSourcesReply = await fetch(eventSourceURL);
+    if (!eventSourcesReply.ok) {
+        console.error(`🛑 Failed to fetch event sources: ${eventSourcesReply.statusText}`);
+
+        return [];
+    }
+
+    const eventSourcesJson = await eventSourcesReply.json();
+
+    return eventSourcesJson.sources;
```

Comment:
The yamcs instance on viper ctrl returns results in the format of:
```
{
  "source": ["a", "b"]
}
```

so this would need to be `eventSourcesJson.source` for that format to work instead of `eventSourcesJson.sources`.

Reply:
ah, this is [deprecated](https://docs.yamcs.org/yamcs-http-api/events/list-event-sources/). i'll check for both to support older versions of YAMCS